targets/TARGET_Samsung/TARGET_SIDK_S5JS100/port_api.c

```diff
@@ -0,0 +1,111 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/* @file  : port_api.c
+ * @brief : GPIO port API source code
+ * @date  : June 2019
+ *
+ * @note  : Add chip dependent feature and control hardware GPIO port
+ *
+ */
+#include "port_api.h"
+#include "pinmap.h"
+#include "gpio_api.h"
+
+PinName port_pin(PortName port, int pin_n)
+{
+    return (PinName)((port << PORT_SHIFT) | pin_n);
+}
+
+void port_init(port_t *obj, PortName port, int mask, PinDirection dir)
+{
+#if 0//revise me
```

Comment:
These function are empty, so does it mean that init won't work?

Reply:
@jh6186,
Could you please answer @bulislaw 's question regarding this point?
https://github.com/ARMmbed/mbed-os/pull/12106/files#diff-8dbe235e8945eb43c1cd541816071223R50-R63

Reply:
Since gpio_dir, gpio_mode API is available for GPIO variation, port API seems not used in this target.

Reply:
Thanks for the tip and I made sure removing the driver code didn't cause compile errors. 9fad25f8dcc30591ce5c5f7c51945eabb8de2617 submitted.

Reply:
@bulislaw and @0xc0170 

Would it be possible to use gpio_dir and gpio_mode internally for port_dir and port_mode functions? Also I do not see the need of having duplicate function sets of gpio_api and port_api HAL layers and raised a question on the forum (https://forums.mbed.com/t/the-difference-between-gpio-api-and-port-api-hal-layers/6658). It would be appreciated if you can give us guidance.

(CC: @jh6186)